services/ui/ws/window_tree.cc

```diff
@@ -929,6 +929,9 @@ void WindowTree::AddExternalModeWindowManagerState(
 }
 
 bool WindowTree::ShouldRouteToWindowManager(const ServerWindow* window) const {
+  if (window_server_->IsInExternalWindowMode())
+    return false;  // No window manager is available in external mode.
+
```

Comment:
so, this used to return FALSE. Maybe something broke.

BTW, ::ShouldRouteToWindowManager calls ::IsWindowCreatedByWindowManager. Suggestion: maybe this early return should actually move to ::IsWindowCreatedByWindowManager, so we would be fixing two methods for external window mode?

::IsWindowCreatedByWindowManager which still has a null-check for window_manager_state. was it added by us? should we remove it?

2211 bool WindowTree::IsWindowCreatedByWindowManager(                                 
2212     const ServerWindow* window) const {                     
2213   // The WindowManager is attached to the root of the Display, if there isn't a
2214   // WindowManager attached, the window manager didn't create this window.
2215   const WindowManagerDisplayRoot* display_root =
2216       GetWindowManagerDisplayRoot(window); 
2217   if (!display_root || !display_root->window_manager_state())
2218     return false;